app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

```diff
@@ -36,9 +41,39 @@ class ExplorationActivityTest {
     }
   }
 
+  @Test
+  fun textExplorationActivity_onBackPressed_showsStopExplorationDialog() {
+    ActivityScenario.launch<ExplorationActivity>(createExplorationActivityIntent(TEST_EXPLORATION_ID_30)).use {
+      pressBack()
+      onView(withText(R.string.stop_exploration_dialog_title))
+    }
+  }
+
+  @Test
+  fun textExplorationActivity_onBackPressed_showsStopExplorationDialog_clickCancel_dismissesDialog() {
+    ActivityScenario.launch<ExplorationActivity>(createExplorationActivityIntent(TEST_EXPLORATION_ID_30)).use {
+      pressBack()
+      onView(withText(R.string.stop_exploration_dialog_title))
+      onView(withText(R.string.stop_exploration_dialog_cancel_button)).inRoot(isDialog()).check(matches(isDisplayed()))
+        .perform(click())
+      it.onActivity { activity ->  not(activity.isFinishing)}
+    }
+  }
+
+  @Test
+  fun textExplorationActivity_onBackPressed_showsStopExplorationDialog_clickLeave_closesExplorationActivity() {
+    ActivityScenario.launch<ExplorationActivity>(createExplorationActivityIntent(TEST_EXPLORATION_ID_30)).use {
+      pressBack()
+      onView(withText(R.string.stop_exploration_dialog_title))
```

Comment:
Ditto here.

Reply:
Removed this. This line was actually not needed.